### PR TITLE
feat: Add pie series percentages

### LINKED ISSUE #2888: Optionally display % values on Pie widget labels

### PR DESCRIPTION
**Pull request by jbrundage**
Also resolved a bug that was preventing Legend from working with 2D charts
Fixes #2888
Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

### CODE REVIEW

**Comment by jbrundage** [comment]
image

File: packages/chart/src/Pie.ts [hunk 1]
```diff
@@ -52,10 +54,18 @@ export class Pie extends SVGWidget {
...
     }
-
+    calcTotalValue(): number {
+        return this.d3Pie(this.data()).reduce((acc, d) => {
+            return acc + d.data[1];
+        }, 0);
+    }
+    getLabelText(d) {
+        const _perc = ((d.data[1] / this._totalValue) * 100).toFixed(1).split(".0").join("");
+        return d.data[0] + (this.showSeriesPercentage() ? `: ${_perc}%` : "");
```
Comment (GordonSmith) [inline packages/chart/src/Pie.ts]: Minor:  no need to calc percentage if its not needed.
Style comment:  For consistency, don't add an "_" to local variables,

File: packages/chart/src/Pie.ts [hunk 2]
```diff
@@ -247,7 +256,11 @@ Pie.prototype._class += " chart_Pie";
 Pie.prototype.implements(I2DChart.prototype);
 Pie.prototype.implements(ITooltip.prototype);
 Pie.prototype.mixin(Utility.SimpleSelectionMixin);
-
+export interface Pie {
+    showSeriesPercentage(): boolean;
+    showSeriesPercentage(_: boolean): this;
+}
+Pie.prototype.publish("showSeriesPercentage", false, "boolean");
```
Comment (GordonSmith) [inline packages/chart/src/Pie.ts]: Add description

File: packages/chart/src/Pie.ts [hunk 3]
```diff
@@ -52,10 +54,22 @@ export class Pie extends SVGWidget {
...
     calcOuterRadius() {
-        const maxTextWidth = this.textSize(this.data().map(d => d[0]), "Verdana", 12).width;
+        const maxTextWidth = this.textSize(this.data().map(d => this.getLabelText({ data: d })), "Verdana", 12).width;
         return Math.min(this._size.width - maxTextWidth * 2 - 20, this._size.height - 12 * 3) / 2 - 2;
     }
-
+    calcTotalValue(): number {
+        return this.d3Pie(this.data()).reduce((acc, d) => {
+            return acc + d.data[1];
+        }, 0);
```
Comment (GordonSmith) [inline packages/chart/src/Pie.ts]: I don't think there is any need to run the data through a d3Pie to simply get a total?

Comment (jbrundage) [reply packages/chart/src/Pie.ts]: @GordonSmith this has been resolved

**Pull request merged**